### PR TITLE
fix: allow cluster bucket export to export single bucket

### PR DESCRIPTION
**Pull request by r-scheele**
## Description
mc command to export single bucket information is not working. It worked a few days ago but now I am getting this error below..
`mc admin cluster bucket export play/abdul`
mc: <ERROR> Unable to download file data. createtemp azprod/rvm-prod-rvm-prod-metadata-: pattern contains path separator.
## How to test this PR?
`./mc admin cluster bucket export play/abdul`

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by harshavardhana** [comment]
> Please check this subnet issue here

@r-scheele we do reference private links in public issues. Ideally open a new GitHub issue if you must redacting private information or simply send a fix.

**Comment by r-scheele** [comment]
Oh, apologies for that. I'll send a fix now

File: cmd/admin-cluster-bucket-export.go [hunk 1]
```diff
@@ -82,29 +84,31 @@ func mainClusterBucketExport(ctx *cli.Context) error {
...
-
 	// Close everything
 	r.Close()
 	tmpFile.Close()
-
 	// We use 4 bytes of the 32 bytes to identify the file.
 	downloadPath := fmt.Sprintf("%s-%s-metadata.%s", aliasedURL, bucket, ext)
+	// Create necessary directories.
+	dir := filepath.Dir(downloadPath)
+	if err := os.MkdirAll(dir, 0o755); err != nil {
```
Comment (harshavardhana) [inline cmd/admin-cluster-bucket-export.go:108]:
```suggestion
	if e := os.MkdirAll(dir, 0o755); e != nil {
```

File: cmd/admin-cluster-bucket-export.go [hunk 2]
```diff
@@ -82,29 +84,31 @@ func mainClusterBucketExport(ctx *cli.Context) error {
...
+		log.Fatal(err)
```
Comment (harshavardhana) [inline cmd/admin-cluster-bucket-export.go:109]:
```suggestion
		fatalIf(probe.NewError(e).Trace(dir))
```

File: cmd/cat-main.go [hunk 3]
```diff
@@ -163,6 +170,8 @@ type catOpts struct {
 
 // parseCatSyntax performs command-line input validation for cat command.
 func parseCatSyntax(ctx *cli.Context) catOpts {
+	// Validate command-line arguments.
+	checkCatSyntax(ctx)
```
Comment (harshavardhana) [inline cmd/cat-main.go:174]:
```suggestion
	checkCatSyntax(ctx)
	
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Comment by harshavardhana** [comment]
Please resolve the conflicts @r-scheele - thanks

**Pull request merged**